ert_shared/ensemble_evaluator/prefect_ensemble/prefect_ensemble.py

```diff
@@ -0,0 +1,322 @@
+                        stage_id=step["stage_id"],
+                        ee_id=ee_id,
+                        on_failure=partial(self._on_task_failure, url=dispatch_url),
+                        run_path=self.config.get("run_path"),
+                        storage_config=self.config.get("storage"),
+                        max_retries=self.config.get("max_retries", 2),
+                        retry_delay=timedelta(seconds=2),
+                    )
+                    result = stage_task(expected_res=inputs)
+
+                    for output in step.get("outputs", []):
+                        output_to_res[output] = result["outputs"]
+        return flow
+
+    def evaluate(self, config, ee_id):
+        self._ee_dispach_url = config.get("dispatch_url")
+        self._ee_id = ee_id
+        self._eval_proc = multiprocessing.Process(
+            target=self._evaluate,
+            args=(self._ee_dispach_url, ee_id),
+        )
+        self._eval_proc.daemon = True
+        self._eval_proc.start()
+
+    def _evaluate(self, dispatch_url, ee_id):
+        sys.stderr = open("prefect.log", "w")
+        input_files = gen_coef(
+            self.config["parameters"], self.config["realizations"], self.config
+        )
+
+        self.store_resources()
+
+        with Client(dispatch_url) as c:
+            event = CloudEvent(
+                {
+                    "type": ids.EVTYPE_ENSEMBLE_STARTED,
+                    "source": f"/ert/ee/{self._ee_id}",
+                    "datacontenttype": "application/json",
+                },
+            )
+            c.send(to_json(event).decode())
+        self.run_flow(ee_id, dispatch_url, input_files)
+
+        with Client(dispatch_url) as c:
+            event = CloudEvent(
+                {
+                    "type": ids.EVTYPE_ENSEMBLE_STOPPED,
+                    "source": f"/ert/ee/{self._ee_id}",
+                    "datacontenttype": "application/json",
+                },
+            )
+            c.send(to_json(event).decode())
+
+    def run_flow(self, ee_id, dispatch_url, input_files):
+        real_per_batch = self.config["max_running"]
+        real_range = range(self.config["realizations"])
+        i = 0
+        while i < self.config["realizations"]:
+            realization_range = real_range[i : i + real_per_batch]
+            flow = self.get_flow(ee_id, dispatch_url, input_files, realization_range)
+            flow.run(executor=_get_executor(self.config["executor"]))
+            i = i + real_per_batch
+
+    def store_resources(self):
+        storage = storage_driver_factory(
+            self.config.get("storage"), self.config.get("config_path", ".")
+        )
+        for stage in self.config["stages"]:
+            for step in stage["steps"]:
+                res_list = []
+                for res in step["resources"]:
+                    res_list.append(storage.store(res))
+                step["resources"] = res_list
```

Comment:
This mutates the config, which is not good. And also we shouldn't use the config object to store state.

Reply:
Very good find! side effects are quite bad.

Reply:
Perhaps the resources should be ready and already resolved in the description given to the ensemble. Since we don't have the Experiment concept yet, perhaps we just keep the storage location in a map somewhere.

Reply:
Think that makes sense, and also move out the parameter sampling :slightly_smiling_face:

Reply:
I already made an issue for the parameter sampling to be moved out https://github.com/equinor/ert/issues/1302